lib/grppcatr.gi

```diff
@@ -466,6 +466,49 @@ end );
 
 #############################################################################
 ##
+#M  MaximalNormalSubgroups( <G> )
+##
+InstallMethod( MaximalNormalSubgroups, "for solvable groups",
+              [ IsGroup ],
+              RankFilter( IsGroup and IsSolvableGroup )
+              - RankFilter( IsGroup ),
+
+  function( G )
+    local Gf,     # FactorGroup of G
+          hom,    # homomorphism from G to Gf
+          MaxGf;  # MaximalNormalSubgroups of Gf
+
+    if 0 in AbelianInvariants(G) then
```

Comment:
So you method is actually called for arbitrary groups (not just solvable ones), and you assume that you can compute AbelianInvaits, IsAbelian, IsSolvableGroup "cheaply". Hmm